app/src/main/java/ch/epfl/sdp/appart/utils/FileIO.java

```diff
@@ -0,0 +1,132 @@
+package ch.epfl.sdp.appart.utils;
+
+import android.graphics.Bitmap;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.ObjectInputStream;
+import java.io.ObjectOutputStream;
+import java.util.Map;
+import java.util.function.Supplier;
+
+/**
+ * This is a util class for file operations.
+ */
+public class FileIO {
+
+    //Prevent the construction of this class, as all the methods are static.
+    private FileIO() {
+    }
+
+    /**
+     * Reads a map object from the path. The Map object has String as keys
+     * and object as values.
+     *
+     * @param path the path to the file that holds the map
+     * @return the map resulting from file reading, or null if the operation
+     * fails
+     */
+    public static Map<String, Object> readMapObject(String path) {
+        FileInputStream fis;
+        Map<String, Object> map;
+        try {
+            fis = new FileInputStream(path);
+            ObjectInputStream ois = new ObjectInputStream(fis);
+            //noinspection unchecked
+            map = (Map<String, Object>) ois.readObject();
+            ois.close();
+            fis.close();
+        } catch (IOException | ClassNotFoundException e) {
+            e.printStackTrace();
+            return null;
+        }
+
+        return map;
+    }
+
+    /**
+     * This method creates a directory pointed to by path, and performs
+     * already exists in case the directory already exists.
+     *
+     * @param path          the path to the directory we want to create
+     * @param alreadyExists function that will be performed in case the
+     *                      directory already exists
+     * @return a boolean that indicates if the operation succeeded or not
+     */
+    public static boolean createFolder(String path,
+                                       Supplier<Boolean> alreadyExists) {
```

Comment:
There are several callbacks type in Java. For instance Function takes one argument and returns a value, Consumer only takes an argument and Supplier doesn't take any argument but return a value. Here, the supplier is used as a callback because we want to be able to do different kind of things if the folder already exists. In the case of the user we want to delete the profile picture and in the case of an ad we want to delete the extra photos. So, the actual deletion will take place "in" the supplier, and it will return if it failed or not.

I hope this is more clear. I will try to improve the documentation for this part.